sdk/Pulumi/Deployment/IDeployment.cs

```diff
@@ -103,6 +103,22 @@ Output<T> Invoke<T>(
             InvokeArgs args,
             InvokeOptions? options = null);
 
+        /// <summary>
+        /// Dynamically invokes the function '<paramref name="token"/>', which is offered by a
+        /// provider plugin.
+        /// <para/>
+        /// The result of <see cref="Invoke{T}(string, InvokeArgs, InvokeOptions)"/> will be a <see cref="Output"/> resolved to the
+        /// result value of the provider plugin.
+        /// <para/>
+        /// The <paramref name="args"/> inputs can be a bag of computed values(including, `T`s,
+        /// <see cref="Task{TResult}"/>s, <see cref="Output{T}"/>s etc.).
+        /// </summary>
+        Output<T> Invoke<T>(
+            string token,
+            InvokeArgs args,
+            InvokeOutputOptions? options = null);
```

Comment:
All the new overloads shouldn't be optional parameters. Otherwise if you call `Invoke(token, args)`, it's ambiguous to the compiler.

```suggestion
            InvokeOutputOptions options);
```

Applies throughout